src/node.cc

```diff
@@ -4419,6 +4419,11 @@ void RunAtExit(Environment* env) {
 }
 
 
+uv_loop_t* GetCurrentEventLoop(v8::Isolate* isolate) {
+  return Environment::GetCurrent(isolate)->event_loop();
```

Comment:
Leaks a handle into the caller's scope.

The overload that takes a `v8::Isolate*` calls `Environment::GetCurrent(isolate->GetCurrentContext())` but that won't work when there is no current context or it doesn't have an associated `Environment` (i.e, non-node context.)

That should be documented because it's the kind of thing that add-ons will run into.  You could robustify (def a word) this function by doing:
```c++
uv_loop_t* GetCurrentEventLoop(v8::Isolate* isolate) {
  HandleScope handle_scope(isolate);
  auto context = isolate->GetCurrentContext();
  if (context.IsEmpty()) return nullptr;
  return Environment::GetCurrent(context);
}
```
That can still crash with a non-node context, though.  Maybe we should file a CL with V8 to expose 
`v8::Context::SlowGetAlignedPointerFromEmbedderData()` in order to safely retrieve the `Environment` pointer.

Reply:
Yeah, thanks for catching those. Updated with your suggestion!